src/catalog/column_catalog.cpp

```diff
@@ -165,28 +168,74 @@ bool ColumnCatalog::InsertColumn(oid_t table_oid,
   auto val6 = type::ValueFactory::GetBooleanValue(is_primary);
   auto val7 = type::ValueFactory::GetBooleanValue(is_not_null);
 
-  tuple->SetValue(ColumnId::TABLE_OID, val0, pool);
-  tuple->SetValue(ColumnId::COLUMN_NAME, val1, pool);
-  tuple->SetValue(ColumnId::COLUMN_ID, val2, pool);
-  tuple->SetValue(ColumnId::COLUMN_OFFSET, val3, pool);
-  tuple->SetValue(ColumnId::COLUMN_TYPE, val4, pool);
-  tuple->SetValue(ColumnId::IS_INLINED, val5, pool);
-  tuple->SetValue(ColumnId::IS_PRIMARY, val6, pool);
-  tuple->SetValue(ColumnId::IS_NOT_NULL, val7, pool);
-
+  auto constant_expr_0 = new expression::ConstantValueExpression(
+    val0);
+  auto constant_expr_1 = new expression::ConstantValueExpression(
+    val1);
+  auto constant_expr_2 = new expression::ConstantValueExpression(
+    val2);
+  auto constant_expr_3 = new expression::ConstantValueExpression(
+    val3);
+  auto constant_expr_4 = new expression::ConstantValueExpression(
+    val4);
+  auto constant_expr_5 = new expression::ConstantValueExpression(
+    val5);
+  auto constant_expr_6 = new expression::ConstantValueExpression(
+    val6);
+  auto constant_expr_7 = new expression::ConstantValueExpression(
+    val7);
+
+  tuples.emplace_back();
+//  tuples.push_back(std::vector<ExpressionPtr>());
+  auto &values = tuples[0];
+
+  values.emplace_back(constant_expr_0);
+  values.emplace_back(constant_expr_1);
+  values.emplace_back(constant_expr_2);
+  values.emplace_back(constant_expr_3);
+  values.emplace_back(constant_expr_4);
+  values.emplace_back(constant_expr_5);
+  values.emplace_back(constant_expr_6);
+  values.emplace_back(constant_expr_7);
   // Insert the tuple
-  return InsertTuple(std::move(tuple), txn);
+  return InsertTupleWithCompiledPlan(&tuples, txn);
 }
 
 bool ColumnCatalog::DeleteColumn(oid_t table_oid,
                                  const std::string &column_name,
                                  concurrency::TransactionContext *txn) {
-  oid_t index_offset =
-      IndexId::PRIMARY_KEY;  // Index of table_oid & column_name
-  std::vector<type::Value> values;
-  values.push_back(type::ValueFactory::GetIntegerValue(table_oid).Copy());
-  values.push_back(
-      type::ValueFactory::GetVarcharValue(column_name, nullptr).Copy());
+  std::vector<oid_t> column_ids(all_column_ids);
+
+  auto *table_oid_expr =
+      new expression::TupleValueExpression(type::TypeId::INTEGER, 0,
+                                           ColumnId::TABLE_OID);
```

Comment:
I don't think the allocated memory has been freed. Irrespective, you should probably use smart pointers instead of working with raw pointers.  The same applies to a bunch of other places in the code where raw pointers could be replaced with smart pointers.